alibi/explainers/anchors/anchor_text.py

```diff
@@ -0,0 +1,556 @@
+####-Changed-#### # noqa
+# Refactored to contain only the AnchorText class.
+#################
+import copy
+import logging
+import string
+from copy import deepcopy
+####-Changed-#### # noqa
+# -from abc import abstractmethod
+# -from functools import partial
+# -from typing import (TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple,
+#                      Type, Union)
+from typing import (TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Union)
+#################
+
+import numpy as np
+import spacy
+####-Changed-#### # noqa
+# -import tensorflow as tf
+#################
+
+
+####-Changed-#### # noqa
+from alibi.utils.missing_optional_dependency import import_optional
+#################
+from alibi.api.defaults import DEFAULT_DATA_ANCHOR, DEFAULT_META_ANCHOR
+from alibi.api.interfaces import Explainer, Explanation
+from alibi.exceptions import (AlibiPredictorCallException,
+                              AlibiPredictorReturnTypeError)
+####-Changed-#### # noqa
+# -from alibi.utils.lang_model import LanguageModel
+from alibi.utils.missing_optional_dependency import MissingDependency
+from alibi.utils import LanguageModel
+#################
+
+from alibi.utils.wrappers import ArgmaxTransformer
+from .anchor_base import AnchorBaseBeam
+from .anchor_explanation import AnchorExplanation
+
+####-Changed-#### # noqa
+from .text_samplers import UnknownSampler, SimilaritySampler, load_spacy_lexeme_prob
+
+LanguageModelSampler = import_optional(
+    'alibi.explainers.anchors.language_model_text_sampler',
+    names=['LanguageModelSampler'])
+#################
+
+if TYPE_CHECKING:
+    import spacy  # noqa: F811
+    ####-Changed-#### # noqa
+    from alibi.explainers.anchors.language_model_text_sampler import LanguageModel as LanguageModelType
```

Comment:
Do we need to do the `as LanguageModelType` here? Seeing as it's imported directly from the implementation and the optional deps are installed in a dev environment?

Reply:
As discussed offline, consider using an explicit `if/else` block and avoid aliasing for greater clarity. Document convention on `CONRIBUTING.md`